src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -955,7 +973,11 @@ private SafeProcessHandle GetProcessHandle(int access, bool throwIfExited)
                         }
                     }
                 }
-                return _processHandle;
+
+                // If we dispose of our contained handle we'll be in a bad state. NetFX dealt with this
+                // by doing a try..finally around every usage of GetProcessHandle and only disposed if
```

Comment:
The only place I see us doing `using` on `GetProcessHandle` is in `WaitForInputIdleCore`. Everywhere else, we do a try around `GetProcessHandle` followed by a finally with `ReleaseProcessHandle`. 

So seems to me, if I understood right, we should either change to `using` everywhere, or do the minimal change which would be to change htat one using back to the dekstop pattern and avoid this change you made here. For post 2.0, we can make it consistently using if we want. ?